openstf/feature_engineering/feature_applicator.py

```diff
@@ -97,8 +107,11 @@ def add_features(self, df: pd.DataFrame, latency_config=None) -> pd.DataFrame:
 
         # NOTE this is required since apply_features could add additional features
         if self.feature_names is not None:
-            # Add horizon to requested features else it is removed
-            features = self.feature_names + ["horizon"]
+            # Add horizon to requested features else it is removed, and if needed the proloaf feature (historic_load)
+            if pj is not None and pj["model"] == "proloaf":
```

Comment:
if pj is not optional you dont have to check wheter it is none. Besides that I dont think this check will work in case pj is none as the evaluation of pj["model"] == "proloaf" will raise an exception